safekeeper/src/timeline.rs

```diff
@@ -364,7 +328,7 @@ impl Timeline {
             (replica_state.remote_consistent_lsn != Lsn::MAX && // Lsn::MAX means that we don't know the latest LSN yet.
              replica_state.remote_consistent_lsn >= shared_state.sk.inmem.commit_lsn);
             if stop {
-                shared_state.deactivate_walsender(&self.zttid);
+                shared_state.active = false;
```

Comment:
That's not a "compute_connected", more like a "pageserver_connected". We can think and do it later.

Reply:
We shouldn't touch `active` here directly, as it depends on s3 offloading as well. `update_status` should be called instead.

Thinking on this, there is one more rough moment -- we could add num computes knowledge to pageserver to force it connecting even when it is caughtup -- it would slightly decrease startup time. But that is also separate.

Reply:
Adjusted that too, but please not that with the `update_status` it's possible to not to shutdown walreceiver which is, code-wise, different from 100% termination that was here before.